lib/email-validation/blacklisted_email.rb

```diff
@@ -16,19 +12,13 @@ class BlacklistedEmail < ::ActiveRecord::Base
 
     before_save :parse_email
 
-    workflow do
-      state :bounce
-      state :unsubscribe
-      state :admin
-    end
-
     ORIGIN_ADMIN = 'admin'
     ORIGIN_BOUNCE = 'bounce'
     ORIGIN_UNSUBSCRIBE = 'unsubscribe'
 
-    def self.exists? email
+    def self.bounce_or_admin? email
       email = BlacklistedEmail.parse_email_address(email.to_s)
-      BlacklistedEmail.where(email: email).exists?
+      BlacklistedEmail.where(email: email, origin: [ORIGIN_BOUNCE, ORIGIN_ADMIN]).exists?
     end
 
     def self.bounced? email
```

Comment:
let's not change simply messaging update into gem refactoring, as we've discussed, the whole functionality needs to be thought through again and probably there's no other way than 'przekopka' ;)